files/scripts/selinux/trivalent/trivalent.if

```diff
@@ -0,0 +1,113 @@
+## <summary>trivalent browser</summary>
+
+########################################
+## <summary>
+##	Execute trivalent in the trivalent domain.
+## </summary>
+## <param name="domain">
+## <summary>
+##	Domain allowed to transition.
+## </summary>
+## </param>
+#
+interface(`trivalent_domtrans',`
+	gen_require(`
+		type trivalent_t, trivalent_exec_t;
+	')
+
+	domtrans_pattern($1, trivalent_exec_t, trivalent_t)
+')
+
+
+########################################
+## <summary>
+##	Execute trivalent in the trivalent domain.
+## </summary>
+## <param name="domain">
+## <summary>
+##	Domain allowed to transition.
+## </summary>
+## </param>
+## <param name="role">
+##	<summary>
+##	Role allowed access.
+##	</summary>
+## </param>
+#
+interface(`trivalent_run',`
+	gen_require(`
+		type trivalent_t;
+        attribute_role trivalent_roles;
+	')
+
+    trivalent_domtrans($1)
+    roleattribute $2 trivalent_roles;
+')
+
+
+
+########################################
+## <summary>
+##	Execute trivalent script in the trivalent domain.
```

Comment:
This comment is wrong - `domtrans_pattern($1, trivalent_script_exec_t, trivalent_script_t)` - it's executing the script in the trivalent script domain, not the trivalent domain